pkg/api/job_runs.go

```diff
@@ -553,6 +572,55 @@ func runTestRunAnalysis(failedTest models.ProwJobRunTest, jobRun *models.ProwJob
 	return analysis, nil
 }
 
+func isHighRiskInOtherPRs(bqc *bigquery.Client, failedTest models.ProwJobRunTest, jobRun *models.ProwJobRun) bool {
+	pr := jobRun.PullRequests[0]
+	endTime := jobRun.Timestamp.Add(jobRun.Duration)
+	if jobRun.Timestamp.IsZero() {
+		endTime = time.Now()
+	}
+	log.Infof("Evaluating if test '%s' is high risk in other PRs for job %s", failedTest.Test.Name, jobRun.ProwJob.Name)
+	_, jobSuffix, found := strings.Cut(jobRun.ProwJob.Name, "pull-ci-"+pr.Org+"-"+pr.Repo)
+	if !found {
+		return false
+	}
+	queryStr := `SELECT COUNT(*) FROM ` +
+		fmt.Sprintf("%s.%s.%s", "openshift-ci-data-analysis", "ci_data_autodl", "risk_analysis_test_results") +
+		fmt.Sprintf(" INNER JOIN %s.%s.%s jobs", "openshift-gce-devel", "ci_analysis_us", "jobs") +
+		` ON JobRunName=jobs.prowjob_build_id` +
+		fmt.Sprintf(" WHERE PartitionTime BETWEEN TIMESTAMP('%s') AND TIMESTAMP('%s') AND", endTime.Add(-12*time.Hour).Format(time.RFC3339), endTime.Add(3*time.Hour).Format(time.RFC3339)) +
+		`  RiskLevel>=100 AND` +
+		fmt.Sprintf("  TestName='%s' AND", failedTest.Test.Name) +
+		fmt.Sprintf("  (org!='%s' OR repo!='%s' OR pr_number!='%d') AND", pr.Org, pr.Repo, pr.Number) +
+		fmt.Sprintf("  prowjob_job_name LIKE '%%%s'", jobSuffix)
+	q := bqc.BQ.Query(queryStr)
+
+	it, err := q.Read(context.TODO())
+	if err != nil {
+		log.WithError(err).Error("Failed querying high risk items from bigquery")
+		return false
+	}
+
+	var rowCount int64
+	for {
+		var values []bqlib.Value
+		err := it.Next(&values)
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			log.WithError(err).Error("error parsing number of high risk items from bigquery")
+			return false
+		}
+		rowCount = values[0].(int64)
+		if rowCount > 0 {
+			log.Infof("High risk items found in other PRs for job %s test '%s'", jobRun.ProwJob.Name, failedTest.Test.Name)
```

Comment:
How about adding the count to the logging?  Maybe even returning the count so we could use it in a metric